src/modules/resource/inventory.c

```diff
@@ -0,0 +1,652 @@
+/************************************************************\
+ * Copyright 2020 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+/* inventory.c - find instance resources by static methods
+ *
+ * Try to obtain instance resources by the following methods in order
+ * of descending precedence:
+ * 1. resources defined by TOML config: [resource] path = "file"
+ * 2. resources defined by R in enclosing Flux instance
+ *
+ * Then commit the result to resource.R in the KVS, and post
+ * resource-define event to resource.eventlog.
+ *
+ * Alternatively, resource.R is pre-loaded into the KVS.
+ * If found, this takes precedence over case 2 but not 1.  This allows
+ * the R product to be reused upon module reload (saving work),
+ * or for resources to be faked in test.
```

Comment:
could just list this as case 2 above, then case 2 above becomes case 3?

Reply:
well I was trying to stay synced with cases 1,2,3 as described in #3238 as that kind of became a shorthand in discussions.  The whole block comment is a bit lacking in detail though.  Let me work it over.